src/main/java/amazin/service/ShoppingCartService.java

```diff
@@ -0,0 +1,86 @@
+package amazin.service;
+
+import amazin.model.Book;
+import amazin.model.Item;
+import amazin.model.ShoppingCart;
+import amazin.model.User;
+import amazin.repository.BookRepository;
+//import amazin.repository.ShoppingCartRepository;
+import amazin.repository.ItemRepository;
+import amazin.repository.ShoppingCartRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.*;
+
+/**
+ * Created by lauramachado on 2019-03-18.
+ */
+@Service
+public class ShoppingCartService {
+
+    @Autowired
+    BookRepository bookRepository;
+
+    @Autowired
+    ShoppingCartRepository cartRepository;
+
+    @Autowired
+    ItemRepository itemRepository;
+
+    public ShoppingCart createCart(User u){
+        ShoppingCart cart = new ShoppingCart();
+        cart.setUser(u);
+        cartRepository.save(cart);
+        return cart;
+    }
+
+    public void addItem(ShoppingCart cart, long bookId){
+        if(cart.itemExists(bookId)){
+            Item i = cart.getItem(bookId);
+            incrementItem(cart, i);
+        }else{
+            Item item = createItem(bookId);
+            if(item != null){
+                cart.addItem(item);
+            }
+        }
+
+        cartRepository.save(cart);
```

Comment:
the cart shouldn't be saved if no new item is added (in the case of having a bookId that doesn't exist)